### PR TITLE
feat:kill compliances 2

### PR DESCRIPTION
**Pull request by SowmyaFyle**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
https://app.clickup.com/1864988/v/l/1rx8w-35684

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export class ExpensesService {
   constructor(@Inject(PAGINATION_SIZE) private paginationSize: number, private spenderService: SpenderService) {}
 
   @Cacheable({
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     cacheBusterObserver: expensesCacheBuster$,
```
Comment (SowmyaFyle) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:23]: eslint fix

Comment (OmkarJ13) [reply src/app/core/services/platform/v1/spender/expenses.service.ts:23]: No way to fix the lint error here?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -4866,36 +4867,64 @@ export class AddEditExpensePage implements OnInit {
...
     }
+
+    this.orgSettingsService
+      .get()
+      .pipe(
+        switchMap((orgSettings) => {
+          const isDuplicateDetectionV2Enabled =
+            orgSettings.duplicate_detection_v2_settings.allowed && orgSettings.duplicate_detection_v2_settings.enabled;
+
+          let duplicateExpenseService$: Observable<{ transaction_ids: string[] }[]>;
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4883]: Should we use the DuplicateSet type here?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -4866,36 +4867,64 @@ export class AddEditExpensePage implements OnInit {
...
+
+          if (isDuplicateDetectionV2Enabled) {
+            duplicateExpenseService$ = this.expensesService.getDuplicatesByExpense(expenseId).pipe(
+              // this will be removed when expenses API is migrated to platform completely
+              map((platformDuplicateSets) =>
+                platformDuplicateSets.map((duplicateSet) => ({ transaction_ids: duplicateSet.expense_ids }))
+              )
+            );
+          } else {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4893]: No way to fix the error?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -4866,36 +4867,64 @@ export class AddEditExpensePage implements OnInit {
...
+      .get()
+      .pipe(
+        switchMap((orgSettings) => {
+          const isDuplicateDetectionV2Enabled =
+            orgSettings.duplicate_detection_v2_settings.allowed && orgSettings.duplicate_detection_v2_settings.enabled;
+
+          let duplicateExpenseService$: Observable<{ transaction_ids: string[] }[]>;
+
+          if (isDuplicateDetectionV2Enabled) {
+            duplicateExpenseService$ = this.expensesService.getDuplicatesByExpense(expenseId).pipe(
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4886]: duplicateExpenses$?

File: app-plugin-custom
Comment (OmkarJ13) [inline app-plugin-custom:1]: What is this 😛

**Review by OmkarJ13 — CHANGES_REQUESTED** [review]
Some comments

File: app-plugin-custom
Comment (SowmyaFyle) [reply app-plugin-custom:1]: no idea how this got added 😅 ..will remove

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 4 above)
Comment (SowmyaFyle) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4886]: we are assigning the service instance to the variable and so given the name `duplicateExpenseService$ `

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (SowmyaFyle) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:4893]: i tried but couldn't

File: src/app/core/services/platform/v1/spender/expenses.service.ts (hunk 1 above)
Comment (SowmyaFyle) [reply src/app/core/services/platform/v1/spender/expenses.service.ts:23]: same here

**Pull request closed without merging**